### PR TITLE
feat/node: limit max number of clients per proxy

### PR DESCRIPTION
**Pull request by maqi**

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
r? @afck

(maidsafe_highfive has picked a reviewer for you, use r? to override)

File: src/peer_manager.rs [hunk 1]
```diff
@@ -46,13 +46,16 @@ const CONNECTING_PEER_TIMEOUT_SECS: u64 = 90;
 const CONNECTED_PEER_TIMEOUT_SECS: u64 = 60;
 /// Time (in seconds) after which a `VotedFor` candidate will be removed.
 const CANDIDATE_ACCEPT_TIMEOUT_SECS: u64 = 60;
+/// Maximum allowed number of clients per proxy node.
+const MAX_NUM_OF_CLIENTS_PER_PROXY: usize = 40;
```
Comment (afck) [inline src/peer_manager.rs]: (I'd omit the `NUM_OF_` for brevity; not important, though.)

File: tests/mock_crust/mod.rs [hunk 2]
```diff
@@ -215,3 +216,33 @@ fn whitelist() {
     let _ = poll_all(&mut nodes, &mut clients);
     expect_next_event!(clients[0], Event::Connected);
 }
+
+#[test]
+// Having multiple clients connect to the same proxy node.
+// Expect fail to connect after reach MAX_NUM_OF_CLIENTS_PER_PROXY,
```
Comment (Fraser999) [inline tests/mock_crust/mod.rs]: `MAX_NUM_OF_CLIENTS_PER_PROXY` got missed in the renaming I guess?

**Comment by maqi** [comment]
closed as included in PR https://github.com/maidsafe/routing/pull/1474

**Pull request closed without merging**